internal/config/config.go

```diff
@@ -1140,6 +1140,11 @@ func selectedNodes(nodes []corev1.Node, selectors []metav1.LabelSelector) (map[s
 	res := make(map[string]bool)
 OUTER:
 	for _, node := range nodes {
+		_, ok := node.Labels[corev1.LabelNodeExcludeBalancers]
```

Comment:
this is the wrong place to add the logic, as the function is "selected nodes".

I'd rather do the filtering in the controller (when we build the resources structure), or on top of config.For and carrying around the filtered node slices.

Reply:
Or, you can even add the filter in the config controller's predicate.

Reply:
How about the latest changes? I prefer to do these filters in the speaker's shouldAnnoce, it's simply and we can easily add logs to see what happed.

Reply:
It works well on my local machine:

```
➜  metallb git:(exclude_nodes) ✗ kubectl describe svc -n l2-6976 external-local-lb | grep nodeAssigned
  Normal  nodeAssigned  24m (x6 over 45m)  metallb-speaker  announcing from node "kind-worker2" with protocol "layer2"

➜  metallb git:(exclude_nodes) ✗ kubectl label nodes kind-worker2 node.kubernetes.io/exclude-from-external-load-balancers=""
node/kind-worker2 labeled
➜  metallb git:(exclude_nodes) ✗ kubectl describe svc -n l2-6976 external-local-lb | grep nodeAssigned                      
  Normal  nodeAssigned  24m (x6 over 45m)  metallb-speaker  announcing from node "kind-worker2" with protocol "layer2"
  Normal  nodeAssigned  3s (x2 over 38m)   metallb-speaker  announcing from node "kind-control-plane" with protocol "layer2"

➜  metallb git:(exclude_nodes) ✗ kubectl label nodes kind-worker2 node.kubernetes.io/exclude-from-external-load-balancers-  
node/kind-worker2 unlabeled
➜  metallb git:(exclude_nodes) ✗ kubectl describe svc -n l2-6976 external-local-lb | grep nodeAssigned                    
  Normal  nodeAssigned  55s (x2 over 39m)  metallb-speaker  announcing from node "kind-control-plane" with protocol "layer2"
  Normal  nodeAssigned  4s (x7 over 46m)   metallb-speaker  announcing from node "kind-worker2" with protocol "layer2"
```